### PR TITLE
Setup Amplitude tracking

### PR DESCRIPTION
**Pull request by manangouhari**
**What does this PR do?**
Setup amplitude tracking.

### CODE REVIEW

**Comment by bhaskarSingh** [comment]
Remove metomic & integrate a pop-up modal showing that we are tracking you.

**Comment by manangouhari** [comment]
Thoughts on the UX? @harsh242 @bhaskarSingh 
image

**Comment by harsh242** [comment]
> Thoughts on the UX? @harsh242 @bhaskarSingh

Looks good, are we also allowing for customization like earlier or just a Yes No?

**Comment by manangouhari** [comment]
> Looks good, are we also allowing for customization like earlier or just a Yes No?
Not allowing customization, not even allowing a Yes No. 
It's there by default, the banner is just an alert of that.

**Comment by manangouhari** [comment]
@bhaskarSingh  can you please check this and let me know if it's good to be merged or anything else needs to be done.

File: src/pages/tezos/academy/index.js [hunk 1]
```diff
@@ -95,6 +104,10 @@ const CurriculumOverview = ({
     return p;
   }, []);
 
+  useEffect(() => {
+    trackEvent('Academy-View');
```
Comment (bhaskarSingh) [inline src/pages/tezos/academy/index.js:108]: You don't have to specifically track a particular view. Amplitude does that by default.

File: src/pages/tezos/customizebot.js [hunk 2]
```diff
@@ -363,6 +364,10 @@ const Customizer = () => {
 
   const [isUser] = useAtom(isUserAtom);
 
+  useEffect(() => {
```
Comment (bhaskarSingh) [inline src/pages/tezos/customizebot.js:367]: No need to specifically track a particular view. Amplitude does that by default.

File: src/pages/tezos/index.js [hunk 3]
```diff
@@ -76,6 +78,10 @@ const TestimonialCard = ({ link, img, name, username, text }) => {
 };
 
 function Landing() {
+  useEffect(() => {
+    trackEvent('Homepage-View');
```
Comment (bhaskarSingh) [inline src/pages/tezos/index.js:82]: No need to specifically track a particular view. Amplitude does that by default.

File: src/pages/tezos/academy/index.js (hunk 1 above)
Comment (bhaskarSingh) [reply src/pages/tezos/academy/index.js:108]: Other than this. Everything looks great.

Comment (manangouhari) [reply src/pages/tezos/academy/index.js:108]: Okay. But afaik, specific views were being tracked manually in the v1 code. 
Any reason behind that?

**Comment by bhaskarSingh** [comment]
It was for testing purpose, I think. I don't remember, honestly.

**Comment by manangouhari** [comment]
> It was for testing purpose, I think. I don't remember, honestly.

Can you please double-check and confirm if the views are being tracked automatically?

**Comment by bhaskarSingh** [comment]
> Can you please double-check and confirm if the views are being tracked automatically?

Just checked. Pageviews are being tracked automatically.

Screenshot 2021-02-26 at 7 22 13 PM

**Comment by manangouhari** [comment]
@bhaskarSingh it's done. Ready to be merged now.

File: src/pages/tezos/academy/index.js [hunk 4]
```diff
@@ -47,12 +51,17 @@ const CourseCard = ({ m, i, progress }) => {
...
             </h3>
           )}
-          <Link
-            to={`/tezos/academy/${m.frontmatter.slug}`}
+          <button
+            onClick={() => {
+              trackEventWithProperties('Module-Click-From-Academy', {
+                slug: m.frontmatter.slug,
+              });
+              navigate(`/tezos/academy/${m.frontmatter.slug}`);
```
Comment (bhaskarSingh) [inline src/pages/tezos/academy/index.js:59]: Throws error `Uncaught ReferenceError: navigate is not defined`

File: src/components/Footer/index.js [hunk 5]
```diff
@@ -45,8 +46,45 @@ const LinkContainer = ({ children }) => {
 };
 
 const Footer = () => {
+  //fix: sync this with localStorage
+  const [cookieBanner, setCookieBanner] = useState(true);
```
Comment (bhaskarSingh) [inline src/components/Footer/index.js:50]: @manangouhari  By default it's true. I think that's why there's a flicker even after closing the pop-up.

To solve the bug. Below logic would help.

```javascript
const [cookieBanner, setCookieBanner] = useState(false); // keep it false by default


  useEffect(() => {
    if (typeof window !== 'undefined') {
      let banner = localStorage.getItem('cookieBanner');
// show if banner has never been closed before.
      if (banner === null) {
        setCookieBanner(true);
      }
    }
  }, []);
```

**Comment by manangouhari** [comment]
> > Can you please double-check and confirm if the views are being tracked automatically?
> 
> Just checked. Pageviews are being tracked automatically.
> 
> Screenshot 2021-02-26 at 7 22 13 PM
> Screenshot 2021-02-26 at 7 34 39 PM

Can't find it in the dashboard for beta.

image

**Pull request merged**